### ISSUE TITLE
Window glitches when dragging app between monitors with different scaling

### ISSUE THREAD

**Issue by Felicity-R** [dotnet/maui#11777]
### Description

Dragging a Maui app between different monitor causes visual glitches and eventually stops responding. See attached video to see what I mean: [screen-capture.webm](https://user-images.githubusercontent.com/62772237/204944972-ed0e29f2-b5b8-465c-905a-96af43d966f7.webm). In the video I'm just recording my 2nd monitor; the laptop screen is below it so that's where the app is when I drag it off screen below. In the end I need to ctrl+alt+delete to get to the task manager and force close it because everything stops responding.

### Steps to Reproduce

1. Have a dual-monitor setup
2. Set custom scaling on monitor in the display settings. e.g. I have my laptop scaling set to 125% and a secondary monitor with scaling at 100%. (If I change the laptop to also be 100% scaling then the bug does not occur).
![image](https://user-images.githubusercontent.com/62772237/204944959-b93861ad-0998-4bdd-a11d-0f73d5775869.png)
4. Create a new app from the .Net MAUI App template (using .Net7)
5. Launch the app
6. Drag the app from one monitor to the other. If it's still behaving normally, drag it back.

Expected results:
You can freely move the app around and it will work fine.

Actual results:
After moving from one monitor to the other a few times, it will break. Sometimes the app window flashes around wildly, sometimes just the edge of the window twitches - see video in description.

### Link to public reproduction project repository

Can repro in the Maui template app

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 10 19043.2251

### Did you find any workaround?

Don't use different scaling

**Comment by drasticactions** [comment]
If you try this in a WinUI application without MAUI, what happens?

**Comment by Felicity-R** [comment]
Good question. I tried with a new app from the "Blank App, Pacakged (WinUI 3 in Desktop)" template, changed it to .net7 to match the Maui project and was not able to reproduce the issue there. I also tried adding a ScrollViewer and some TextBoxes there and still no repro.
Then I pared down the Maui app, removing all the controls from the main page so it was just a blank page, and the Maui app still had the issue.

**Comment by DerPate2010** [comment]
I see the same behavior on my machine. One display is at 200% and a second one at 150%. I see different kinds of flickering effects and the content seems to toggle between both DPI settings (= sizes).

Based on the behavior my assumption is, that the MAUI window is doing some resizing strategy before the user releases the mouse cursor on the second screen

1. It seems as if the window is being resized as a reaction to the message from Windows when moving to the second screen. 
2. As a result of the resizing the main part of the smaller window is now back on the first screen. => 
3. Windows is sending a message to scale to this size. 
4. Maui rescales to the larger window and the main part of the window is on the second screen again => 
5. goto 1

When I break the execution I always see stack trace like this:

<img width="1210" alt="image" src="https://user-images.githubusercontent.com/6896463/206151817-7f17f5e2-a697-46eb-894e-70e248bcf99e.png">

**Comment by LucasFelixAquino** [comment]
I have the same issue with Maui Blazor App. any update?

**Comment by drasticactions** [comment]
https://github.com/dotnet/maui/issues/14597#issuecomment-1509985432

This mentions the possible issue. Calculating the Width, Height, X, and Y could be causing it to redraw and freak out.

**Comment by salarcode** [comment]
Addition to my comment https://github.com/dotnet/maui/issues/14597#issuecomment-1509985432
If I add this line to the method, we can see this method is in hot path
```
if (_batchFrameUpdate > 0)
{
	Debug.WriteLine($"_batchFrameUpdate={_batchFrameUpdate}, frame={frame}");
}
```
Like this:
![image](https://user-images.githubusercontent.com/1272095/232495371-e714cc9d-e68f-415d-a09d-eee18130e720.png)


I will get this when I move MAUI window from a monitor with 150% scaling to another one with 100% scaling:
```
_batchFrameUpdate=1, frame={X=-279 Y=160 Width=797 Height=1211}
_batchFrameUpdate=2, frame={X=-279 Y=160 Width=797 Height=807}
_batchFrameUpdate=2, frame={X=-186 Y=106.66666412353516 Width=797.3333129882812 Height=1210.6666259765625}
_batchFrameUpdate=3, frame={X=-186 Y=160 Width=797.3333129882812 Height=1210.6666259765625}
_batchFrameUpdate=1, frame={X=-519 Y=151 Width=797 Height=1211}
_batchFrameUpdate=2, frame={X=-519 Y=151 Width=797 Height=807}
_batchFrameUpdate=2, frame={X=-519 Y=151 Width=797 Height=1211}
_batchFrameUpdate=1, frame={X=-519 Y=151 Width=531 Height=1211}
_batchFrameUpdate=1, frame={X=-519 Y=151 Width=531 Height=807}
```

It seems to me that updating bindable properties in such hot path is not a good idea.


.

**Comment by salarcode** [comment]
https://github.com/dotnet/maui/blob/48d02949c511bd979f061343f90ea63cd227558e/src/Controls/src/Core/HandlerImpl/Window/Window.Impl.cs#L209-L212
These lines are causing performance regression imo


![image](https://user-images.githubusercontent.com/1272095/233478964-8a26c9f3-3a36-4a20-bb22-30e6ce942bee.png)

@jonathanpeppers pinging for help

**Comment by jonathanpeppers** [comment]
The picture of a stack trace above shows this is called:

https://github.com/dotnet/maui/blob/2b823f8503748f115028e671b4fff9048628c462/src/Core/src/Platform/Windows/WindowExtensions.cs#L26

Hoping this was profiled in `Release` mode, but can you share this file here?

So, is this what is happening?

1. User drags window
2. Windows APIs -> tells MAUI the position changed
3. MAUI `OnPropertyChanged()` -> calls Windows API to update position
4. Repeat in a loop?

> These lines are causing performance regression imo

Regression from what? .NET 6?

**Comment by salarcode** [comment]
Unfortunately i didn't save that file, wasn't expecting it be needed. And yes it was in Release mode.

Here is another run I did in Release mode. 
What I did is just run `Maui.Controls.Sample` and move the window between two monitors with different scaling as I've mentioned before.
At the end I managed to glitch the app and it was stuck and unresponsive like this :


https://user-images.githubusercontent.com/1272095/233505719-065d6299-ccab-4c41-844a-5c8943d68260.mp4


Here is the profile session for above :

[Report20230421-0905.zip](https://github.com/dotnet/maui/files/11291047/Report20230421-0905.zip)

> Regression from what? .NET 6?

I think based on #4942 yes.
